sycl/doc/extensions/proposed/sycl_ext_oneapi_non_uniform_groups.asciidoc

```diff
@@ -0,0 +1,1161 @@
+
+  linear_id_type get_group_linear_range() const;
+
+  linear_id_type get_local_linear_range() const;
+
+  bool leader() const;
+};
+
+}
+----
+
+[source,c++]
+----
+id_type get_group_id() const;
+----
+_Returns_: An `id` representing the index of the tangle-group.
+
+NOTE: This will always be an `id` with all values set to 0, since there can
+only be one tangle-group.
+
+[source,c++]
+----
+id_type get_local_id() const;
+----
+_Returns_: An `id` representing the calling work-item's position within
+the tangle-group.
+
+[source,c++]
+----
+range_type get_group_range() const;
+----
+_Returns_: A `range` representing the number of tangle-groups.
+
+NOTE: This will always return a `range` of 1 as there can only be one
+tangle-group.
+
+[source,c++]
+----
+range_type get_local_range() const;
+----
+_Returns_: A `range` representing the number of work-items in the tangle-group.
+
+[source,c++]
+----
+id_type get_group_linear_id() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_group_id()`.
+
+[source,c++]
+----
+id_type get_local_linear_id() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_local_linear_id()`.
+
+[source,c++]
+----
+range_type get_group_linear_range() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_group_range()`.
+
+[source,c++]
+----
+range_type get_local_linear_range() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_local_range()`.
+
+[source,c++]
+----
+bool leader() const;
+----
+_Returns_: `true` for exactly one work-item in the tangle-group, if the calling
+work-item is the leader of the tangle-group, and `false` for all other
+work-items in the tangle-group. The leader of the tangle-group is guaranteed to
+be the work-item for which `get_local_id()` returns 0.
+
+
+==== Usage Examples
+
+A `tangle_group` can be used in conjunction with constructs like loops and
+branches to safely communicate between all work-items executing the same
+control flow.
+
+NOTE: This differs from a `ballot_group` because a `tangle_group` requires the
+implementation to track group membership. Which group type to use will depend
+on a combination of implementation/backend/device and programmer preference.
+
+[source, c++]
+----
+auto sg = it.get_sub_group();
+
+auto will_branch = sg.get_local_linear_id() % 2 == 0;
+if (will_branch)
+{
+  // synchronize across the work-items that took the branch
+  auto inner = sycl::ext::oneapi::experimental::get_tangle_group(sg);
+  sycl::group_barrier(inner);
+
+  // reduce across subset of outer work-items that took the branch
+  float ix = sycl::reduce_over_group(inner, x, plus<>());
+
+  // further subdivide group into work-items that will branch further
+  auto will_branch_further = inner.get_local_linear_id() < 8;
+  if (will_branch_further)
+  {
+    // synchronize across the work-items that took both branches
+    auto inner_inner = sycl::ext::oneapi::experimental::get_tangle_group(inner);
```

Comment:
Same comment here as earlier ... I think this line is not supported in the current version of this extension.

Reply:
Also removed in 536e29441.